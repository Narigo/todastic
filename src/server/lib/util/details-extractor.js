import { extractLabels } from "./label-extractor.js";

export function extractDetails(inputString) {
  const { labels, text } = extractLabels(inputString);
  const labelsWithoutTrackedTime = labels.filter(label => !/track/i.test(label.name));
  const trackedTimes = labels
    .filter(label => /track/i.test(label.name))
    .reduce((trackedTimes, label) => [...trackedTimes, ...label.args], []);

  return { text, labels: labelsWithoutTrackedTime, trackedTimes };
}
