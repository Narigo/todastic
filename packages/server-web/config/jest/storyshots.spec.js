import path from "path";
import util from "util";
import initStoryshots from "@storybook/addon-storyshots";
import registerRequireContext from "babel-plugin-require-context-hook/register";

// To find all the stories, storybook uses require.context(). The
// following line replaces this in test with another function, using
// the Node APIs to find files.
registerRequireContext();

// We need to catch proptype warnings in the tests.

/* eslint-disable no-console */
const consoleWarn = console.warn;
const consoleError = console.error;
const setConsoles = (warn, error) => () => {
  console.warn = warn;
  console.error = error;
};
beforeEach(setConsoles(logToError, logToError));
afterEach(setConsoles(consoleWarn, consoleError));
/* eslint-enable no-console */

function logToError(...args) {
  throw new Error(util.format(...args));
}

initStoryshots({
  configPath: path.resolve(__dirname, "../storybook/config.js")
});
