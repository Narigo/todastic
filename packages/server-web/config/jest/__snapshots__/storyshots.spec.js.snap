// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots AssetImage logo 1`] = `
<svg
  version="1.1"
  viewBox="0 0 151 33"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <use
    xlink:href="#todastic-logo.svg"
  />
</svg>
`;

exports[`Storyshots DropZone default 1`] = `
<div
  class=""
>
  <div>
    <span>
      click here to create a new todo!
    </span>
  </div>
</div>
`;

exports[`Storyshots DropZone dragging 1`] = `
<div
  class="undefined"
>
  <div>
    <span>
      drag here to move!
    </span>
  </div>
</div>
`;

exports[`Storyshots LoginForm failing 1`] = `
<div>
  <div>
    <div
      class="root"
    >
      <svg
        version="1.1"
        viewBox="0 0 151 33"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <use
          xlink:href="#todastic-logo.svg"
        />
      </svg>
    </div>
     
    <form>
      <div>
        <input
          autocomplete="username"
          name="username"
          placeholder="Username"
          type="text"
        />
         
        <button
          type="submit"
        >
          Login
        </button>
         
        <input
          autocomplete="current-password"
          name="password"
          placeholder="Password"
          type="password"
        />
         
        <!---->
      </div>
    </form>
  </div>
</div>
`;

exports[`Storyshots LoginForm succeeding 1`] = `
<div>
  <div>
    <div
      class="root"
    >
      <svg
        version="1.1"
        viewBox="0 0 151 33"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <use
          xlink:href="#todastic-logo.svg"
        />
      </svg>
    </div>
     
    <form>
      <div>
        <input
          autocomplete="username"
          name="username"
          placeholder="Username"
          type="text"
        />
         
        <button
          type="submit"
        >
          Login
        </button>
         
        <input
          autocomplete="current-password"
          name="password"
          placeholder="Password"
          type="password"
        />
         
        <!---->
      </div>
    </form>
  </div>
</div>
`;

exports[`Storyshots TodasticIcon Calendar 1`] = `
<svg
  class="root"
  version="1.1"
  viewBox="0 0 48 48"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <use
    xlink:href="#calendar.svg"
  />
</svg>
`;

exports[`Storyshots TodasticIcon Clock 1`] = `
<svg
  class="root"
  version="1.1"
  viewBox="0 0 48 48"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <use
    xlink:href="#clock.svg"
  />
</svg>
`;

exports[`Storyshots TodasticIcon block 1`] = `
<svg
  class="root block"
  version="1.1"
  viewBox="0 0 48 48"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <use
    xlink:href="#calendar.svg"
  />
</svg>
`;

exports[`Storyshots TodasticIcon inline text 1`] = `
<div>
  There is a 
  <svg
    class="root"
    version="1.1"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <use
      xlink:href="#calendar.svg"
    />
  </svg>
   in my sentence!
</div>
`;

exports[`Storyshots TodasticLogo default 1`] = `
<div
  class="root"
>
  <svg
    version="1.1"
    viewBox="0 0 151 33"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <use
      xlink:href="#todastic-logo.svg"
    />
  </svg>
</div>
`;

exports[`Storyshots TodoItem done 1`] = `
<div
  class="root"
>
  <div
    class="expander"
  >
    <svg
      class="root"
      version="1.1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlink:href="#chevron.svg"
      />
    </svg>
  </div>
   
  <div
    class="undefined status"
  />
   
  <div
    class="textLabels"
  >
    <div
      class="root text"
    >
      <div>
        this is a done todo
      </div>
       
      <!---->
    </div>
     
    <ul
      class="root labels"
    />
  </div>
   
  <div
    class="root isExpanded expandedText"
  >
    <div>
      this is a done todo
    </div>
     
    <div />
  </div>
</div>
`;

exports[`Storyshots TodoItem example 1`] = `
<div
  class="root"
>
  <div
    class="expander"
  >
    <svg
      class="root"
      version="1.1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlink:href="#chevron.svg"
      />
    </svg>
  </div>
   
  <div
    class="status"
  />
   
  <div
    class="textLabels"
  >
    <div
      class="root text"
    >
      <div>
        some really nice todo
      </div>
       
      <!---->
    </div>
     
    <ul
      class="root labels"
    >
      <li
        class="label"
      >
        foo
      </li>
      <li
        class="label"
      >
        bar
      </li>
      <li
        class="label"
      >
        baz
      </li>
      <li
        class="label"
      >
        something
      </li>
    </ul>
  </div>
   
  <div
    class="root isExpanded expandedText"
  >
    <div>
      some really nice todo
    </div>
     
    <div>
      <p>
        Bacon ipsum dolor amet chuck meatloaf tongue fatback capicola. Pork leberkas shank pork chop rump. Cow doner pork belly jerky. Pork chop alcatra drumstick spare ribs strip steak brisket burgdoggen turkey leberkas boudin prosciutto. Doner pig biltong alcatra hamburger tongue salami corned beef frankfurter leberkas picanha. Filet mignon hamburger rump short ribs.
      </p>
      

      <p>
        T-bone meatball venison beef ribs flank chuck ground round, swine leberkas short loin biltong ham hock landjaeger strip steak. Bacon short loin sirloin flank. Filet mignon kevin bresaola bacon. Drumstick pancetta cupim ground round rump, swine bacon venison shank t-bone leberkas pork kevin. Drumstick jowl fatback, shankle spare ribs alcatra swine prosciutto. Pork loin pork turducken venison, frankfurter ham hock turkey kielbasa alcatra tenderloin shoulder ham landjaeger salami sausage. Prosciutto pork chop biltong chuck salami flank jowl tail picanha jerky shankle.
      </p>
      

      <p>
        Alcatra ham shankle beef ribs shank ham hock pork chop burgdoggen short ribs sirloin kevin andouille pork bacon. Beef brisket prosciutto frankfurter andouille tenderloin alcatra ribeye venison short ribs. Hamburger landjaeger bacon, pork belly turducken andouille pancetta ham leberkas. Beef ribs fatback andouille tenderloin. Filet mignon flank pancetta swine alcatra pork loin porchetta doner kielbasa shank tongue fatback cow drumstick andouille. Boudin burgdoggen alcatra frankfurter. Meatloaf shank ham leberkas kevin ball tip.
      </p>
      

      <p>
        Jerky prosciutto kielbasa rump pastrami burgdoggen spare ribs beef ribs pancetta. Filet mignon t-bone tongue, boudin turducken hamburger frankfurter jerky ham ham hock shankle cow kevin shoulder. Rump swine short loin ball tip, shank tri-tip filet mignon corned beef landjaeger hamburger beef ribs ham hock beef biltong. Shoulder pastrami turkey chicken andouille spare ribs. Ball tip cupim doner boudin chicken, sirloin meatball pork loin landjaeger filet mignon t-bone jowl hamburger pork chop andouille. Andouille tongue kielbasa shankle spare ribs, venison t-bone.
      </p>
      

      <p>
        Jerky tail jowl flank kevin. Hamburger jowl biltong boudin pig alcatra jerky turkey bresaola spare ribs fatback porchetta. Meatball pork chop boudin, cupim brisket beef bacon flank pancetta beef ribs spare ribs drumstick strip steak picanha ball tip. Tongue picanha cupim short loin chuck ball tip shank. Pork jowl jerky doner swine. Shank boudin burgdoggen, short loin pork chop venison corned beef capicola.
      </p>
      

    </div>
  </div>
</div>
`;

exports[`Storyshots TodoItem open 1`] = `
<div
  class="root"
>
  <div
    class="expander"
  >
    <svg
      class="root"
      version="1.1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlink:href="#chevron.svg"
      />
    </svg>
  </div>
   
  <div
    class="status"
  />
   
  <div
    class="textLabels"
  >
    <div
      class="root text"
    >
      <div>
        this is an open todo
      </div>
       
      <!---->
    </div>
     
    <ul
      class="root labels"
    />
  </div>
   
  <div
    class="root isExpanded expandedText"
  >
    <div>
      this is an open todo
    </div>
     
    <div />
  </div>
</div>
`;

exports[`Storyshots TodoItem with labels 1`] = `
<div
  class="root"
>
  <div
    class="expander"
  >
    <svg
      class="root"
      version="1.1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlink:href="#chevron.svg"
      />
    </svg>
  </div>
   
  <div
    class="status"
  />
   
  <div
    class="textLabels"
  >
    <div
      class="root text"
    >
      <div>
        this is a todo with labels
      </div>
       
      <!---->
    </div>
     
    <ul
      class="root labels"
    >
      <li
        class="label"
      >
        foo
      </li>
      <li
        class="label"
      >
        bar
      </li>
      <li
        class="label"
      >
        baz
      </li>
      <li
        class="label"
      >
        something
      </li>
    </ul>
  </div>
   
  <div
    class="root isExpanded expandedText"
  >
    <div>
      this is a todo with labels
    </div>
     
    <div />
  </div>
</div>
`;

exports[`Storyshots TodoItem/TodoItemExpander default 1`] = `
<div
  class=""
>
  <svg
    class="root"
    version="1.1"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <use
      xlink:href="#chevron.svg"
    />
  </svg>
</div>
`;

exports[`Storyshots TodoItem/TodoItemLabels multiple labels 1`] = `
<ul
  class="root"
>
  <li
    class="label"
  >
    hello
  </li>
  <li
    class="label"
  >
    foo
  </li>
  <li
    class="label"
  >
    bar
  </li>
</ul>
`;

exports[`Storyshots TodoItem/TodoItemLabels no labels 1`] = `
<ul
  class="root"
/>
`;

exports[`Storyshots TodoItem/TodoItemLabels single label 1`] = `
<ul
  class="root"
>
  <li
    class="label"
  >
    hello
  </li>
</ul>
`;

exports[`Storyshots TodoItem/TodoItemStatus done 1`] = `
<div
  class="undefined"
/>
`;

exports[`Storyshots TodoItem/TodoItemStatus open 1`] = `
<div
  class=""
/>
`;

exports[`Storyshots TodoItem/TodoItemText expanded 1`] = `
<div
  class="root isExpanded"
>
  <div>
    # hello
  </div>
   
  <div>
    <p>
      this is a lot of
    </p>
    

    <p>
      TEXT!
    </p>
    

  </div>
</div>
`;

exports[`Storyshots TodoItem/TodoItemText markdown 1`] = `
<div
  class="root"
>
  <div>
    # hello
  </div>
   
  <!---->
</div>
`;

exports[`Storyshots TodoItem/TodoItemText some text 1`] = `
<div
  class="root"
>
  <div>
    hello
  </div>
   
  <!---->
</div>
`;
